src/pybind/mgr/prometheus/module.py

```diff
@@ -291,6 +307,24 @@ def get_df(self):
                                     pool['stats'][stat],
                                     (pool['id'],))
 
+    def get_fs(self):
+        fs_map = self.get('fs_map')
+        active_daemons = []
+        for fs in fs_map['filesystems']:
+            # collect fs metadata
+            data_pools = ",".join([str(pool) for pool in fs['mdsmap']['data_pools']])
+            self.metrics.append('fs_metadata', 1,
+                                (data_pools,
+                                 fs['id'],
+                                 fs['mdsmap']['max_mds'],
+                                 fs['mdsmap']['metadata_pool'],
+                                 fs['mdsmap']['fs_name']))
+            active_daemons = [daemon for gid, daemon in
+                           fs['mdsmap']['info'].items()]
+        for daemon in fs_map['standbys'] + active_daemons:
+            self.metrics.append('mds_metadata', 1,
```

Comment:
This will generate extra series when daemons transition from standby to active -- I'm not sure we really want/need to mention standbys in the metadata at all?  They don't output any meaningful perf counters to look up.

Reply:
I would suggest leaving data_pools in, but taking max_mds out.

data pools are rarely changed, and usually only explicitly by the administrator.  max_mds is something we're likely to be auto-adjusting dynamically in the future.

Reply:
Also, everywhere we output rank for an MDS, need to also include the filesystem ID, as the rank is not globally unique (all filesystems have a rank 0,1,2 etc)